src/middlewares/asyncWrapper.js

```diff
@@ -0,0 +1,23 @@
+
+/**
+  * A wrapper for express js controller for error handling
+  *
+  * @function
+  *
+  * @param {Function} wrappedFunction - the main controller
+  *
+  * @return {Function} - callback that execute the controller
+  */
+export default (wrappedFunction) => {
+  const currentFunction = wrappedFunction;
+
+  return async (request, response, next) => {
+    try {
+      const { status, data, message = '' } = await currentFunction(request, response, next);
+
+      return response.status(status).json({ status: 'success', data, message });
```

Comment:
You can use optional object properties here instead of sending "message" back as an empty string i.e. 
```suggestion
      return response.status(status).json({ status: 'success', data, ...(message && { message }) });
```
This way, if message is undefined. Nothing gets added to the response object. You'll also have to remove the default property on line 16 i.e.

```javascript
const { status, data, message } = await currentFunction(request, response, next);
```